stdlib/public/Concurrency/DispatchGlobalExecutor.inc

```diff
@@ -247,7 +248,10 @@ static void swift_task_enqueueGlobalWithDelayImpl(JobDelay delay,
   job->SchedulerPrivate[Job::DispatchQueueIndex] =
       DISPATCH_QUEUE_GLOBAL_EXECUTOR;
 
-  dispatch_time_t when = dispatch_time(DISPATCH_TIME_NOW, delay);
+  // dispatch_time expects a signed int64, though we have an unsigned delay
+  // so we must convert accordingly.
+  int64_t dispatchDelay = delayUnsigned & INT64_MAX;
```

Comment:
so since UInt64.max is seen as -1 Int64, we truncate here; Dispatch expects a signed integer in dispatch_time so we can't pass an uint as we'll get wrong interpretation